morethuente.go

```diff
@@ -0,0 +1,383 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import "math"
+
+// MoreThuente is a Linesearcher that finds steps that satisfy both the
+// sufficient decrease and curvature conditions (the strong Wolfe conditions).
+//
+// References:
+//  - More, J.J. and D.J. Thuente: Line Search Algorithms with Guaranteed Sufficient
+//    Decrease. ACM Transactions on Mathematical Software 20(3) (1994), 286-307
+type MoreThuente struct {
+	// DecreaseFactor is the constant factor in the sufficient decrease
+	// (Armijo) condition.
+	// It must be from the interval [0, 1). The default value is 0.
+	DecreaseFactor float64
+	// CurvatureFactor is the constant factor in the Wolfe conditions. Smaller
+	// values result in a more exact line search.
+	// It must be from the interval [0, 1). If it is zero, it will be defaulted
+	// to 0.9.
+	CurvatureFactor float64
+	// StepTolerance is the tolerance for an acceptable step. The line search
```

Comment:
This is too vague. "StepTolerance sets the minimum acceptable width for the linesearch interval. If the relative interval length is less than this value, ErrLinesearchFailure is returned.